### PR TITLE
Fix deadcode issues

### PR DESCRIPTION
**Pull request by mjtrangoni**
Hi @bergquist,

Related to #10381, I fixed all [deadcode](https://github.com/tsenart/deadcode) issues and added this check to CircleCI. See,

```
build.go:26:1:warning: versionRe is unused (deadcode)
build.go:47:1:warning: minGoVersion is unused (deadcode)
build.go:329:1:warning: verifyGitRepoIsClean is unused (deadcode)
build.go:355:1:warning: ChangeWorkingDir is unused (deadcode)
build.go:495:1:warning: buildArch is unused (deadcode)
build.go:503:1:warning: run is unused (deadcode)
pkg/api/annotations.go:298:1:warning: canSaveByRegionID is unused (deadcode)
pkg/middleware/render_auth.go:34:1:warning: renderContextFunc is unused (deadcode)
pkg/services/sqlstore/migrations/stats_mig.go:5:1:warning: addStatsMigrations is unused (deadcode)
```

Tell me if I should restore some code and just comment it out instead.

### CODE REVIEW

File: pkg/services/sqlstore/migrations/stats_mig.go [hunk 1]
```diff
@@ -2,38 +2,6 @@ package migrations
 
 import . "github.com/grafana/grafana/pkg/services/sqlstore/migrator"
 
-func addStatsMigrations(mg *Migrator) {
```
Comment (bergquist) [inline pkg/services/sqlstore/migrations/stats_mig.go]: please comment this code instead of deleting.

Comment (bergquist) [inline pkg/services/sqlstore/migrations/stats_mig.go]: Comment this code instead of deleting it. It's being used in some development but not released yet.

Comment (mjtrangoni) [reply pkg/services/sqlstore/migrations/stats_mig.go]: @bergquist done, PTAL

**Comment by bergquist** [comment]
Thank you for contributing!

**Pull request merged**